### PR TITLE
feat: Add an Operator Selector to data-research page 

### PR DESCRIPTION
**Pull request by yardenmezi**
Add the selection option of operators on the research page
image

### CODE REVIEW

File: src/model/operator.ts [hunk 1]
```diff
@@ -5,7 +5,18 @@ export type Operator = {
...
+export const getRelevantOperators = async (onlyRelevantOperators: boolean) => {
+  const agencyList = await getAgencyList()
+  if (!onlyRelevantOperators) {
+    const allOperators = Array.from(
+      agencyList.map((agency) => ({
+        name: agency.agency_name,
+        id: agency.operator_ref.toString(),
+      })),
+    )
+    allOperators.push({ name: 'הכל', id: '' })
```
Comment (NoamGaash) [inline src/model/operator.ts:17]: why do we need the `הכל` here?
```suggestion
    const allOperators = agencyList.map((agency) => ({
        name: agency.agency_name,
        id: agency.operator_ref.toString(),
      }))
    allOperators.push({ name: 'הכל', id: '' })
```

File: src/pages/components/OperatorSelector.tsx [hunk 2]
```diff
@@ -3,34 +3,41 @@ import { useTranslation } from 'react-i18next'
 import { Operator, getRelevantOperators } from 'src/model/operator'
 import { Autocomplete, TextField } from '@mui/material'
 
+export enum FilterOperatorOptions {
+  ALL,
+  RELEVANT,
+  MAJOR,
+}
+
```
Comment (NoamGaash) [inline src/pages/components/OperatorSelector.tsx:11]: :+1:

File: src/pages/components/OperatorSelector.tsx [hunk 3]
```diff
@@ -3,34 +3,41 @@ import { useTranslation } from 'react-i18next'
...
   setOperatorId,
-  onlyMajorOperators = false,
+  filter = FilterOperatorOptions.RELEVANT
 }: OperatorSelectorProps) => {
   const { t } = useTranslation()
   const [operators, setOperators] = useState<Operator[]>([])
   useEffect(() => {
     const majorOperatorsIds = ['3', '5', '15', '18', '25', '34']
-    getRelevantOperators().then((resultObj) =>
+    getRelevantOperators(filter != FilterOperatorOptions.ALL).then((resultObj) =>
```
Comment (NoamGaash) [inline src/pages/components/OperatorSelector.tsx:27]:
```suggestion
    getRelevantOperators(filter != FilterOperatorOptions.ALL)
    .then(list => filter === FilterOperatorOptions.ALL ? [...list, {id: '', name: t('operatorSelector.all')}])
    .then((resultObj) =>
```

**Review by NoamGaash — CHANGES_REQUESTED** [review]
Thanks for your contribution!
The code looks great- just please try to separate the data layer from the UI layer. having `all` come from the data getter could be confusing for future developers

File: src/model/operator.ts (hunk 1 above)
Comment (yardenmezi) [reply src/model/operator.ts]: Changed, Thanks! (assumed the problem with adding this in this function doesn't fit the function name?)

**Review by NoamGaash — APPROVED** [review]
Looks great! thanks for adding a story

File: src/model/operator.ts (hunk 1 above)
Comment (NoamGaash) [reply src/model/operator.ts]: Thanks! It's also about the data model should not be aware of the UI options

**Pull request merged**